src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -495,6 +495,18 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
                 return true;
             }
 
+            // No space before an asterisk that's part of a PointerTypeSyntax.
+            if (next.IsKind(SyntaxKind.AsteriskToken) && next.Parent is PointerTypeSyntax)
+            {
+                return false;
+            }
+
+            // The last asterisk of a pointer declaration should be followed by a space.
+            if (token.IsKind(SyntaxKind.AsteriskToken) && token.Parent is PointerTypeSyntax && next.IsKind(SyntaxKind.IdentifierToken))
+            {
+                return true;
+            }
```

Comment:
this check seems strange.  Don't we always want this for the identifier of a decl?  wouldn't it be cleaner to just say `if (next.IsKind(SyntaxKind.Identifier) && next.Parent.IsKind(VariableDeclarator)) return true;`?

Reply:
@CyrusNajmabadi That will change the behavior if the pointer type is the return type of a method. We want a space after the asterisk in this case, and it is not a variable declarator.

Reply:
that won't work if the following token is not an identifier.  for example: `int* this[int x, int y]`.

Reply:
@CyrusNajmabadi Good case! Are there any other cases other than indexers?